waldo/data/validate.go

```diff
@@ -28,9 +41,21 @@ func ValidateUploadToken(token string) error {
 		return errors.New("Empty upload token")
 	}
 
-	if uploadTokenRE.FindString(token) != token {
+	if userTokenRE.FindString(token) != token {
 		return fmt.Errorf("Invalid upload token syntax: %q", token)
 	}
 
 	return nil
 }
+
+func ValidateUserToken(token string) error {
+	if len(token) == 0 {
+		return errors.New("Empty user token")
+	}
+
+	if appTokenRE.FindString(token) != token && userTokenRE.FindString(token) != token {
```

Comment:
Hmmm. I will look at `Match` to see if it is a better fit for the purpose.

Reply:
I’m guess you mean `MatchString` (since `Match` takes a `[]byte` argument)? Then I could write:
```
if !appTokenRE.MatchString(token) && !userTokenRE.MatchString(token) {
}
```
Yes?

Reply:
Yup